### PR TITLE
Spawn-based execution

### PR DESCRIPTION
**Pull request by freiksenet**

### CODE REVIEW

File: python_modules/dagit/dagit/pipeline_execution_manager.py [hunk 1]
```diff
@@ -107,8 +108,17 @@ def __init__(self, error_info):
 
 class MultiprocessingExecutionManager(PipelineExecutionManager):
     def __init__(self):
-        self._processes_lock = multiprocessing.Lock()
+        if hasattr(multiprocessing, 'get_context'):
```
Comment (schrockn) [inline python_modules/dagit/dagit/pipeline_execution_manager.py]: explanatory comment plz about everything at play here

File: python_modules/dagit/dagit/pipeline_execution_manager.py [hunk 2]
```diff
@@ -107,8 +108,17 @@ def __init__(self, error_info):
...
+            self._context = multiprocessing.get_context('spawn')
+        else:
+            # Python 2.7 doesn't support alternative starting methods
+            self._context = multiprocessing
```
Comment (schrockn) [inline python_modules/dagit/dagit/pipeline_execution_manager.py]: can we rename "_context" to "_multiprocessing_context"

File: python_modules/dagit/dagit/app.py [hunk 3]
```diff
@@ -62,6 +64,10 @@ def repository(self):
     def error(self):
         return self.repo_error
 
+    @property
```
Comment (schrockn) [inline python_modules/dagit/dagit/app.py]: this doesn't seem to add too much value

File: python_modules/dagit/dagit/pipeline_run_storage.py [hunk 4]
```diff
@@ -131,7 +132,9 @@ def __init__(self, pipeline_run, start_cursor=None):
 
     def __call__(self, observer):
         self.observer = observer
-        self.observer.on_next(self.pipeline_run.logs_after(self.start_cursor))
+        events = self.pipeline_run.logs_after(self.start_cursor)
```
Comment (schrockn) [inline python_modules/dagit/dagit/pipeline_run_storage.py]: what is going on here

File: python_modules/dagit/dagit/app.py (hunk 3 above)
Comment (freiksenet) [reply python_modules/dagit/dagit/app.py]: Just following the surrounding style.

File: python_modules/dagit/dagit/pipeline_run_storage.py (hunk 4 above)
Comment (freiksenet) [reply python_modules/dagit/dagit/pipeline_run_storage.py]: Don't send empty list as an event.

Comment (schrockn) [reply python_modules/dagit/dagit/pipeline_run_storage.py]: oh i missed something i thought this was a bigger change

**Pull request merged**